### PR TITLE
feat(github): Add span evidence to performance issues

### PR DESCRIPTION
**Pull request by ceorourke**
Add span evidence to the description of a GitHub issue created from a performance issue. Currently the GitHub issue is fairly empty as for an error issue it shows the stacktrace, but for a performance issue it's just a link back to the Sentry issue.

Screen Shot 2022-11-04 at 2 03 29 PM

### CODE REVIEW

File: src/sentry/integrations/github/issues.py [hunk 1]
```diff
@@ -19,6 +22,34 @@ def get_issue_url(self, key: str) -> str:
...
+            parent_span,
+            num_repeating_spans,
+            repeating_spans,
+        ) = self.get_performance_issue_description_data(event)
+
+        body = "|  |  |\n"
+        body += "| ------------- | --------------- |\n"
+        body += f"| **Transaction Name** | {truncatechars(transaction_name, 50)} |\n"
+        body += f"| **Parent Span** | {truncatechars(parent_span, 50)} |\n"
+        body += f"| **Repeating Spans ({num_repeating_spans})** | {truncatechars(repeating_spans, 50)} |"
```
Comment (rjo100) [inline src/sentry/integrations/github/issues.py:37]: can we make these `50` a global?

File: src/sentry/integrations/jira/integration.py [hunk 2]
```diff
@@ -331,26 +326,18 @@ def get_persisted_user_default_config_fields(self):
...
+            parent_span,
+            num_repeating_spans,
+            repeating_spans,
+        ) = self.get_performance_issue_description_data(event)
+
+        body = f"| *Transaction Name* | {truncatechars(transaction_name, 50)} |\n"
+        body += f"| *Parent Span* | {truncatechars(parent_span, 50)} |\n"
         body += (
-            f"| *Repeating Spans ({num_repeating_spans})* | {self.truncate_data(repeating_spans)} |"
+            f"| *Repeating Spans ({num_repeating_spans})* | {truncatechars(repeating_spans, 50)} |"
```
Comment (rjo100) [inline src/sentry/integrations/jira/integration.py:340]: can we make these `50` a global?

File: tests/sentry/integrations/github/test_issues.py [hunk 3]
```diff
@@ -96,6 +101,52 @@ def test_create_issue(self, mock_get_jwt):
         payload = json.loads(request.body)
         assert payload == {"body": "This is the description", "assignee": None, "title": "hello"}
 
+    def test_performance_issues_description(self):
+        """Test that a GitHub issue created from a performance issue has span evidence data in it's description"""
```
Comment (rjo100) [inline tests/sentry/integrations/github/test_issues.py:105]: nit: `its`

File: tests/sentry/integrations/github/test_issues.py [hunk 4]
```diff
@@ -96,6 +101,52 @@ def test_create_issue(self, mock_get_jwt):
...
+        description = GitHubIssueBasic().get_group_description(event.group, event)
+        assert "db - SELECT `books_author`.`id`, `books_author" in description
+        title = GitHubIssueBasic().get_group_title(event.group, event)
+        assert (
+            title
+            == 'N+1 Query: SELECT "books_author"."id", "books_author"."name" FROM "books_author" WHERE "books_author"."id" = %s LIMIT 21'
+        )
+
+    def test_error_issues_description(self):
+        """Test that a GitHub issue created from an error issue has message  data in it's description"""
```
Comment (rjo100) [inline tests/sentry/integrations/github/test_issues.py:137]: nit: `its`

**Review by rjo100 — APPROVED** [review]
any way to add a jira test as well in like `tests/sentry/integrations/jira/test_integration.py`? or do we not have any issue tests for jira analogous to the github tests

**Comment by ceorourke** [comment]
> any way to add a jira test as well in like `tests/sentry/integrations/jira/test_integration.py`? or do we not have any issue tests for jira analogous to the github tests

The Jira tests are a little different but were added in the previous PR: https://github.com/getsentry/sentry/pull/40535/files

**Pull request merged**